gobblin-cluster/src/main/java/gobblin/cluster/ClusterEventMetadataGenerator.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.cluster;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+
+import gobblin.annotation.Alias;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.metrics.event.TimingEvent;
+import gobblin.runtime.JobContext;
+import gobblin.runtime.TaskState;
+import gobblin.runtime.api.EventMetadataGenerator;
+
+
+/**
+ * {@link EventMetadataGenerator} that outputs the processed message count and error messages
+ * used for job status tracking
+ */
+@Alias("cluster")
+public class ClusterEventMetadataGenerator implements EventMetadataGenerator{
+  public static final String TASK_FAILURE_MESSAGE_KEY = "task.failure.message";
+
+  private JobContext jobContext;
+
+  public ClusterEventMetadataGenerator(JobContext jobContext) {
+    this.jobContext = jobContext;
+  }
+
+  public Map<String, String> getMetadata(String eventName) {
+    switch (eventName) {
+      case TimingEvent.LauncherTimings.JOB_COMPLETE:
+        return ImmutableMap.of("processedCount", Long.toString(getProcessedCount()));
+      case TimingEvent.LauncherTimings.JOB_FAILED:
+        return ImmutableMap.of("message", getTaskFailureExceptions());
+      default:
+        break;
+    }
+
+    return ImmutableMap.of();
+  }
+
+  /**
+   * Get the number of records written by all the writers
+   * @return Sum of the writer records written count across all tasks
+   */
+  private long getProcessedCount() {
```

Comment:
This is a suggestion that doesn't need to be done now, but might be useful to have eventually:

Various `EventMetadataGenerator`s might be interested in emitting records processed count for example, so it might be useful to refactor this into a separate class that can be called from any generator. Similarly for the task failure exception.